pygmt/tests/test_nearneighbor.py

```diff
@@ -0,0 +1,115 @@
+"""
+Tests for nearneighbor.
+"""
+import os
+
+import numpy.testing as npt
+import pytest
+import xarray as xr
+from pygmt import nearneighbor
+from pygmt.datasets import load_sample_bathymetry
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import GMTTempFile, data_kind
+
+
+@pytest.fixture(scope="module", name="ship_data")
+def fixture_ship_data():
+    """
+    Load the data from the sample bathymetry dataset.
+    """
+    return load_sample_bathymetry()
+
+
+def test_nearneighbor_input_file():
+    """
+    Run nearneighbor by passing in a filename.
+    """
+    output = nearneighbor(
+        data="@tut_ship.xyz", spacing="5m", region=[245, 255, 20, 30], search_radius="10m"
```

Comment:
Need to break the line here to make the `black` linter happy.

```suggestion
        data="@tut_ship.xyz",
        spacing="5m",
        region=[245, 255, 20, 30],
        search_radius="10m",
```